koku/api/provider/provider_manager.py

```diff
@@ -107,6 +108,27 @@ def get_previous_month_data_exists(self):
             completed_datetime__isnull=False,
         ).exists()
 
+    def get_state(self):
+        """Get latest manifest state."""
+        states = {
+            ManifestStep.DOWNLOAD: "pending",
+            ManifestStep.PROCESSING: "pending",
+            ManifestStep.SUMMARY: "pending",
+        }
+        manifests = CostUsageReportManifest.objects.filter(
+            provider=self._uuid,
+            billing_period_start_datetime=self.date_helper.this_month_start,
+        )
+        if manifests:
+            manifest = manifests.latest("creation_datetime")
+            for key in states.keys():
+                if manifest.state.get(key):
+                    if manifest.state.get(key).get("start"):
+                        states[key] = "in-progress"
+                    if manifest.state.get(key).get("end"):
+                        states[key] = "complete"
```

Comment:
It looks like `ManifestState` could use a couple more values for in-progress and complete.

Since `creation_datetime` is nullable, those values should be filtered out before getting the latest record.

If I understand the goal of this method, it looks like we can avoid setting the value in `states` twice by first checking for `ManifestState.END`.

IMO, using a try/except makes this more readable.

```suggestion
        """Get latest manifest state."""
        states = {
            ManifestStep.DOWNLOAD: ManifestState.PENDING,
            ManifestStep.PROCESSING: ManifestState.PENDING,
            ManifestStep.SUMMARY: ManifestState.PENDING,
        }
        try:
            manifest = CostUsageReportManifest.objects.filter(
                provider=self._uuid,
                billing_period_start_datetime=self.date_helper.this_month_start,
            ).filter(creation_datetime__isnull=False).latest("creation_datetime")
        except DoesNotExist:
            return states
            
        for key in states:
            if current_state := manifest.state.get(key):
                if current_state.get(ManifestState.END):
                    states[key] = ManifestState.COMPLETE
                elif current_state.get(ManifestState.START):
                    states[key] = ManifestState.IN_PROGRESS
                    
        return states
```